# Summer 2022 Data Science Intern Challenge - Shopify

**Question 1:** Given some sample data, write a program to answer the following: click here to access the required data set

On Shopify, we have exactly 100 sneaker shops, and each of these shops sells only one model of shoe. We want to do some analysis of the average order value (AOV). When we look at orders data over a 30 day window, we naively calculate an AOV of $3145.13. Given that we know these shops are selling sneakers, a relatively affordable item, something seems wrong with our analysis.

a.	Think about what could be going wrong with our calculation. Think about a better way to evaluate this data.

We first import pandas and read our Excel file as a dataframe.

In [1]:
import pandas as pd

In [2]:
df = pd.read_excel('2019 Winter Data Science Intern Challenge Data Set.xlsx')
df.head()

order_id  shop_id  user_id  order_amount  total_items payment_method  \
0         1       53      746           224            2           cash   
1         2       92      925            90            1           cash   
2         3       44      861           144            1           cash   
3         4       18      935           156            1    credit_card   
4         5       18      883           156            1    credit_card   

               created_at  
0 2017-03-13 12:36:56.190  
1 2017-03-03 17:38:51.999  
2 2017-03-14 04:23:55.595  
3 2017-03-26 12:43:36.649  
4 2017-03-01 04:35:10.773

Let us now sort the order amounts (field 'order_amount') in descending order.

In [3]:
df.sort_values('order_amount', ascending=False)[:20]

order_id  shop_id  user_id  order_amount  total_items payment_method  \
2153      2154       42      607        704000         2000    credit_card   
3332      3333       42      607        704000         2000    credit_card   
520        521       42      607        704000         2000    credit_card   
1602      1603       42      607        704000         2000    credit_card   
60          61       42      607        704000         2000    credit_card   
2835      2836       42      607        704000         2000    credit_card   
4646      4647       42      607        704000         2000    credit_card   
2297      2298       42      607        704000         2000    credit_card   
1436      1437       42      607        704000         2000    credit_card   
4882      4883       42      607        704000         2000    credit_card   
4056      4057       42      607        704000         2000    credit_card   
15          16       42      607        704000         2000    credit_card   
1104      1105       42      607        704000         2000    credit_card   
1562      1563       42      607        704000         2000    credit_card   
2969      2970       42      607        704000         2000    credit_card   
4868      4869       42      607        704000         2000    credit_card   
1362      1363       42      607        704000         2000    credit_card   
691        692       78      878        154350            6          debit   
2492      2493       78      834        102900            4          debit   
3724      3725       78      766         77175            3    credit_card   

                  created_at  
2153 2017-03-12 04:00:00.000  
3332 2017-03-24 04:00:00.000  
520  2017-03-02 04:00:00.000  
1602 2017-03-17 04:00:00.000  
60   2017-03-04 04:00:00.000  
2835 2017-03-28 04:00:00.000  
4646 2017-03-02 04:00:00.000  
2297 2017-03-07 04:00:00.000  
1436 2017-03-11 04:00:00.000  
4882 2017-03-25 04:00:00.000  
4056 2017-03-28 04:00:00.000  
15   2017-03-07 04:00:00.000  
1104 2017-03-24 04:00:00.000  
1562 2017-03-19 04:00:00.000  
2969 2017-03-28 04:00:00.000  
4868 2017-03-22 04:00:00.000  
1362 2017-03-15 04:00:00.000  
691  2017-03-27 22:51:43.203  
2492 2017-03-04 04:37:33.848  
3724 2017-03-16 14:13:25.868

If we look at the order_amount field, we observe that there are many orders with incredibly large value. The largest order is worth **\$704,000** and had **2000** items purchased. This order is repeated **17 times** on different days, all billed at around 4 pm. These could be bulk orders by some other store buying sneakers for reselling them. The next highest order is for **six** items and is worth **\$154,350**, and so on. These orders might be from an affluent customer buying top range products.

We observe that there are certain orders like the above that *do not represent the typical customer*. Such orders may be considered **outliers**. In our analysis, we are taking the average value of all the orders, including the ones for the outliers. Since these outliers are skewing our data towards a higher average order value (AOV), it is also affecting our metric. Thus this metric clearly does not represent a typical daily customer.

A metric that we should use to evaluate this data must not be affected by such outliers.

b.	What metric would you report for this dataset?

Since we want our metric to represent a typical customer and not be affected by incredibly large orders, we can use a median order value instead. A median is a measure of central tendency that describes the value at the center, or at the 50% percentile of the entire dataset. In this case, considering the median order value would be appropriate as it would be the order value of the centermost order and is not skewed by any outliers, in this case bulk orders or orders from affluent customers. A median would therefore more accurately represent a typical order value.

In [4]:
df.median()['order_amount']

<ipython-input-4-39e71209791e>:1: FutureWarning: DataFrame.mean and DataFrame.median with numeric_only=None will include datetime64 and datetime64tz columns in a future version.
  df.median()['order_amount']


284.0

c.	What is its value?

The median order value is **\$284**. This is much more realistic for sneaker orders!